src/web/utils/AppVisibilityUtils.ts

```diff
@@ -0,0 +1,96 @@
+/**
+* AppVisibilityUtils.ts
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT license.
+*
+* Web-specific helpers for firing focus/activity related events
+*/
+
+import SubscribableEvent from 'subscribableevent';
+
+const idleTimeInMs = 10 * 1000;
+
+export class AppVisibilityUtils {
+    private _isIdle = false;
+    private _timers: Array<number> = [];
```

Comment:
Why are you allocating multiple timers? Isn't a single timer sufficient?

Reply:
Hmm i reused the logic from ifvisible. But your right, since we clear it before everytime we allocate it - i can just use one.